xds/internal/balancer/clusterimpl/clusterimpl.go

```diff
@@ -247,8 +247,22 @@ func (b *clusterImplBalancer) UpdateClientConnState(s balancer.ClientConnState)
 	}
 
 	if b.config == nil || b.config.ChildPolicy.Name != newConfig.ChildPolicy.Name {
-		if err := b.child.SwitchTo(bb); err != nil {
-			return fmt.Errorf("error switching to child of type %q: %v", newConfig.ChildPolicy.Name, err)
+		if b.config == nil || b.config.ChildPolicy.Name != newConfig.ChildPolicy.Name {
```

Comment:
We don't need to check this condition anymore. We can blindly build the config for the gracefulswitch balancer and call its `ParseConfig` method.

Reply:
Done